src/fides/api/api/v1/endpoints/user_endpoints.py

```diff
@@ -591,39 +609,64 @@ def user_login(
     )
 
 
-def perform_login(
-    db: Session,
-    client_id_byte_length: int,
-    client_secret_btye_length: int,
-    user: FidesUser,
-) -> ClientDetail:
-    """Performs a login by updating the FidesUser instance and creating and returning
-    an associated ClientDetail.
+async def verify_invite_code(
+    username: str,
+    invite_code: str,
+    db: Session = Depends(get_db),
+) -> FidesUserInvite:
+    """
+    Security dependency to verify the invite code.
+    Returns the validated FidesUserInvite if all the checks pass.
     """
+    user_invite = FidesUserInvite.get_by(db, field="username", value=username)
 
-    client = user.client
-    if not client:
-        logger.info("Creating client for login")
-        client, _ = ClientDetail.create_client_and_secret(
-            db,
-            client_id_byte_length,
-            client_secret_btye_length,
-            scopes=[],  # type: ignore
-            roles=user.permissions.roles,  # type: ignore
-            systems=user.system_ids,  # type: ignore
-            user_id=user.id,
+    if not user_invite:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail="User not found.",
         )
-    else:
-        # Refresh the client just in case - for example, scopes and roles were added via the db directly.
-        client.roles = user.permissions.roles  # type: ignore
-        client.systems = user.system_ids  # type: ignore
-        client.save(db)
 
-    if not user.permissions.roles and not user.systems:  # type: ignore
-        logger.warning("User {} needs roles or systems to login.", user.id)
-        raise AuthorizationError(detail="Not Authorized for this action")
+    if not user_invite.invite_code_valid(invite_code):
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail="Invite code is invalid.",
+        )
 
-    user.last_login_at = datetime.utcnow()
-    user.save(db)
+    if user_invite.is_expired():
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail="Invite code has expired.",
+        )
 
-    return client
+    return user_invite
+
+
+@router.post(
+    urls.USER_ACCEPT_INVITE,
+)
+def accept_user_invite(
```

Comment:
do we have rate limiting on this endpoint? Even though this isn't a public endpoint, just want to ensure we're being extra careful for security here.